weave/trace_server/trace_server_interface.py

```diff
@@ -2085,95 +1350,13 @@ def evaluate_model(self, req: EvaluateModelReq) -> EvaluateModelRes: ...
-    def evaluation_create_v2(
-        self, req: EvaluationCreateV2Req
-    ) -> EvaluationCreateV2Res: ...
-    def evaluation_read_v2(self, req: EvaluationReadV2Req) -> EvaluationReadV2Res: ...
-    def evaluation_list_v2(
-        self, req: EvaluationListV2Req
-    ) -> Iterator[EvaluationReadV2Res]: ...
-    def evaluation_delete_v2(
-        self, req: EvaluationDeleteV2Req
-    ) -> EvaluationDeleteV2Res: ...
-
-    # Models
-    def model_create_v2(self, req: ModelCreateV2Req) -> ModelCreateV2Res: ...
-    def model_read_v2(self, req: ModelReadV2Req) -> ModelReadV2Res: ...
-    def model_list_v2(self, req: ModelListV2Req) -> Iterator[ModelReadV2Res]: ...
-    def model_delete_v2(self, req: ModelDeleteV2Req) -> ModelDeleteV2Res: ...
-
-    # Evaluation Runs
-    def evaluation_run_create_v2(
-        self, req: EvaluationRunCreateV2Req
-    ) -> EvaluationRunCreateV2Res: ...
-    def evaluation_run_read_v2(
-        self, req: EvaluationRunReadV2Req
-    ) -> EvaluationRunReadV2Res: ...
-    def evaluation_run_list_v2(
-        self, req: EvaluationRunListV2Req
-    ) -> Iterator[EvaluationRunReadV2Res]: ...
-    def evaluation_run_delete_v2(
-        self, req: EvaluationRunDeleteV2Req
-    ) -> EvaluationRunDeleteV2Res: ...
-    def evaluation_run_finish_v2(
-        self, req: EvaluationRunFinishV2Req
-    ) -> EvaluationRunFinishV2Res: ...
-
-    # Predictions
-    def prediction_create_v2(
-        self, req: PredictionCreateV2Req
-    ) -> PredictionCreateV2Res: ...
-    def prediction_read_v2(self, req: PredictionReadV2Req) -> PredictionReadV2Res: ...
-    def prediction_list_v2(
-        self, req: PredictionListV2Req
-    ) -> Iterator[PredictionReadV2Res]: ...
-    def prediction_delete_v2(
-        self, req: PredictionDeleteV2Req
-    ) -> PredictionDeleteV2Res: ...
-    def prediction_finish_v2(
-        self, req: PredictionFinishV2Req
-    ) -> PredictionFinishV2Res: ...
-
-    # Scores
-    def score_create_v2(self, req: ScoreCreateV2Req) -> ScoreCreateV2Res: ...
-    def score_read_v2(self, req: ScoreReadV2Req) -> ScoreReadV2Res: ...
-    def score_list_v2(self, req: ScoreListV2Req) -> Iterator[ScoreReadV2Res]: ...
-    def score_delete_v2(self, req: ScoreDeleteV2Req) -> ScoreDeleteV2Res: ...
-
-
-class FullTraceServerInterface(TraceServerInterface, TraceServerInterfaceV2, Protocol):
-    """Complete trace server interface supporting both V1 and V2 APIs.
+class FullTraceServerInterface(TraceServerInterface, Protocol):
+    """Complete trace server interface supporting both V1 and Object APIs.
 
     This protocol represents a trace server implementation that supports the full
-    set of APIs - both legacy V1 endpoints and modern V2 endpoints. Use this type
+    set of APIs - both legacy V1 endpoints and modern Object endpoints. Use this type
     for implementations that need to support both API versions.
     """
 
-    pass
+    @cached_property
+    def object_interface(self) -> oi.ObjectInterface: ...
```

Comment:
curious why you chose to go this route as opposed to the mixin

Reply:
But it seems like we don't expose this on the actual trace server... I wonder why this was implemented at all?  cc: @gtarpenning

Reply:
it doesn't seem to me that this would justify the extra indirection